### ISSUE TITLE
snapshot to s3 not working

### ISSUE THREAD

**Issue by janavenkat** [rancher/rke#1328]
Apparently, there is no option for s3 in the rancher/rke-tools:v0.1.15 docker images when we enabled the backup config in cluster template 

Regarding this pull request https://github.com/rancher/rke/pull/1036 its for ```rke etcd snapshot-save``` quite confusing 

**RKE version:**
v0.2.2

**Docker version: (`docker version`,`docker info` preferred)**

Docker version 18.09.5, build e8ff056

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**

NAME="Ubuntu"
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Openstack
Installed on HA

**cluster.yml file:**
added etcd service section 

```   services:
  etcd:
    extra_args: {}
    extra_binds: []
    extra_env: []
    external_urls: []
    ca_cert: ""
    cert: ""
    key: ""
    path: ""
    snapshot: true
    retention: "24h"
    creation: "5m0s"
    backup_config:
      interval_hours: 1
      retention: 6
      s3backupconfig:
        access_key: *******
        secret_key: *********************
        bucket_name: ****************
        region: *******
        endpoint: ********** 

```

**Steps to Reproduce:**
1. add the above etcd service in cluster.yml
2. rke up
3. after the deployment the pod will be restarting state. Try to ```docker inspect etcd-rolling-snapshots```  and ``` docker logs  etcd-rolling-snapshots ```

```[
    {
        "Id": "044ffe3bb6b85f04aad7940f4ea6613b5a824177b4ec833707929688d8d2a463",
        "Created": "2019-05-06T15:40:01.837517356Z",
        "Path": "/opt/rke-tools/rke-etcd-backup",
        "Args": [
            "rolling-backup",
            "save",
            "--cacert",
            "/etc/kubernetes/ssl/kube-ca.pem",
            "--cert",
            "/etc/kubernetes/ssl/kube-node.pem",
            "--key",
            "/etc/kubernetes/ssl/kube-node-key.pem",
            "--name",
            "etcd-rolling-snapshots",
            "--endpoints=192.168.0.17:2379",
            "--creation=1h",
            "--retention=6h",
            "--s3-backup=true",
            "--s3-endpoint=*******************",
            "--s3-accessKey=*****************",
            "--s3-secretKey=*******************",
            "--s3-bucketName=**********",
            "--s3-region=**************"
        ],
        "State": {
            "Status": "restarting",
            "Running": true,
            "Paused": false,
            "Restarting": true,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2019-05-06T15:46:49.756982841Z",
            "FinishedAt": "2019-05-06T15:46:49.854850814Z"
        },
        "Image": "sha256:c822ad0496a43f399f3bda9f53a723ece115cb52105944a594f52f9234ab63e3",
        "ResolvConfPath": "/var/lib/docker/containers/044ffe3bb6b85f04aad7940f4ea6613b5a824177b4ec833707929688d8d2a463/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/044ffe3bb6b85f04aad7940f4ea6613b5a824177b4ec833707929688d8d2a463/hostname",
        "HostsPath": "/var/lib/docker/containers/044ffe3bb6b85f04aad7940f4ea6613b5a824177b4ec833707929688d8d2a463/hosts",
        "LogPath": "/var/lib/docker/containers/044ffe3bb6b85f04aad7940f4ea6613b5a824177b4ec833707929688d8d2a463/044ffe3bb6b85f04aad7940f4ea6613b5a824177b4ec833707929688d8d2a463-json.log",
        "Name": "/etcd-rolling-snapshots",
        "RestartCount": 16,
        "Driver": "aufs",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/opt/rke/etcd-snapshots/:/backup:z",
                "/etc/kubernetes:/etc/kubernetes:z"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "host",
            "PortBindings": null,
            "RestartPolicy": {
                "Name": "always",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "shareable",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": null,
            "Name": "aufs"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/opt/rke/etcd-snapshots",
                "Destination": "/backup",
                "Mode": "z",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/etc/kubernetes",
                "Destination": "/etc/kubernetes",
                "Mode": "z",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "volume",
                "Name": "483de0db9b38b7ed98ea64ec0538f99bd077dadc968a5852643814ea6e55fb29",
                "Source": "/var/lib/docker/volumes/483de0db9b38b7ed98ea64ec0538f99bd077dadc968a5852643814ea6e55fb29/_data",
                "Destination": "/opt/rke-tools",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
        "Config": {
            "Hostname": "master-2",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "80/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "NGINX_VERSION=1.14.0"
            ],
            "Cmd": [
                "/opt/rke-tools/rke-etcd-backup",
                "rolling-backup",
                "save",
                "--cacert",
                "/etc/kubernetes/ssl/kube-ca.pem",
                "--cert",
                "/etc/kubernetes/ssl/kube-node.pem",
                "--key",
                "/etc/kubernetes/ssl/kube-node-key.pem",
                "--name",
                "etcd-rolling-snapshots",
                "--endpoints=192.168.0.17:2379",
                "--creation=1h",
                "--retention=6h",
                "--s3-backup=true",
                "--s3-endpoint=*****",
                "--s3-accessKey=*********",
                "--s3-secretKey=***********",
                "--s3-bucketName=**********",
                "--s3-region=************"
            ],
            "Image": "rancher/rke-tools:v0.1.15",
            "Volumes": {
                "/opt/rke-tools": {}
            },
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "maintainer": "Rancher Labs <support@rancher.com>"
            },
            "StopSignal": "SIGTERM"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "78af30df8d2d42b5db66030c7542f6fe88ff36f440cd940d73a9c1325ada0686",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/default",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "host": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "442612e556e38666dbd80b9c27b7b9b72a1a6e1589b951457930c1799ff2c109",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Results:**

Docker logs 

```NAME:
   rke-etcd-backup rolling-backup - Perform rolling backups

USAGE:
   rke-etcd-backup rolling-backup [command options] [arguments...]

OPTIONS:
   --endpoints value  Etcd endpoints (default: "127.0.0.1:2379")
   --creation value   Create backups after this time interval in minutes (default: 5m0s)
   --retention value  Retain backups within this time interval in hours (default: 24h0m0s)
   --debug            Verbose logging information for debugging purposes [$RANCHER_DEBUG]
   --once             Take backup only once
   --name value       Backup name to take once
   --cacert value     Etcd CA client certificate path [$ETCD_CACERT]
   --cert value       Etcd client certificate path [$ETCD_CERT]
   --key value        Etcd client key path [$ETCD_KEY]
   
Incorrect Usage.
```

**Comment by janeczku** [comment]
Backup to S3 requires at least `rke-tools` v0.1.17 ([Commit](https://github.com/rancher/rke-tools/commit/db0e7f9369f535ee2d28bf647525e8a200e3c6e9)) but RKE v0.2.2 uses `rke-tools` v0.1.15 for K8s version 1.11.x:

```
~ jan$ rke -v
rke version v0.2.2
~ jan$ rke config --system-images --version v1.11.9-rancher1-1
INFO[0000] Generating images list for version [v1.11.9-rancher1-1]: 
rancher/coreos-etcd:v3.2.18
rancher/rke-tools:v0.1.15
...
```

**Comment by deniseschannon** [comment]
We should update that the versions shipped in v0.2 at least use the rke-tools to be able to snapshot to s3.

**Comment by alena1108** [comment]
Can be tested with rke v0.2.3-rc3

**Comment by sowmyav27** [comment]
Reproduced this issue with `rke:v0.2.2` and k8s: `v1.11.9-rancher1-1`
```# docker logs etcd-rolling-snapshots
Incorrect Usage.

NAME:
   rke-etcd-backup rolling-backup - Perform rolling backups

USAGE:
   rke-etcd-backup rolling-backup [command options] [arguments...]

OPTIONS:
   --endpoints value  Etcd endpoints (default: "127.0.0.1:2379")
   --creation value   Create backups after this time interval in minutes (default: 5m0s)
   --retention value  Retain backups within this time interval in hours (default: 24h0m0s)
   --debug            Verbose logging information for debugging purposes [$RANCHER_DEBUG]
   --once             Take backup only once
   --name value       Backup name to take once
   --cacert value     Etcd CA client certificate path [$ETCD_CACERT]
   --cert value       Etcd client certificate path [$ETCD_CERT]
   --key value        Etcd client key path [$ETCD_KEY]
   
Incorrect Usage.
```
Validated using `rke v0.2.3-rc3`
Created a 1 node cluster with s3, kubernetes_version: `v1.11.9-rancher1-2`.
rke-tools:v0.1.28 is available on cluster. 
No error seen while doing `docker logs etcd-rolling-snapshots`
```
# docker logs etcd-rolling-snapshots
time="2019-05-21T21:37:58Z" level=info msg="Initializing Rolling Backups" creation=1h0m0s retention=6h0m0s 
time="2019-05-21T21:37:58Z" level=info msg="invoking set s3 service client" 
```

**Comment by oxr463** [comment]
@alena1108 were you able to find a workaround?

**Comment by sowmyav27** [comment]
@oxr463 This is fixed in rke version `rke v0.2.3-rc3` and above.
